src/loaders.ts

```diff
@@ -1,282 +1,359 @@
+    ok: boolean; // False if no result is found
 }
 
-const emptyContractResult : ContractResult = {
-  ok: false,
+const emptyContractResult: ContractResult = {
+    ok: false,
 
-  abi: [],
-  name: null,
-  evmVersion: "",
-  compilerVersion: "",
-  runs: 0,
+    abi: [],
+    name: null,
+    evmVersion: "",
+    compilerVersion: "",
+    runs: 0,
 }
 
 export interface ABILoader {
-  loadABI(address: string): Promise<any[]>;
-  getContract(address: string): Promise<ContractResult>
+    loadABI(address: string): Promise<any[]>;
+    getContract(address: string): Promise<ContractResult>
 }
 
 // Load ABIs from multiple providers until a result is found.
 export class MultiABILoader implements ABILoader {
-  loaders: ABILoader[];
-
-  constructor(loaders: ABILoader[]) {
-    this.loaders = loaders;
-  }
-
-  async getContract(address: string): Promise<ContractResult> {
-    for (const loader of this.loaders) {
-      try {
-        const r = await loader.getContract(address);
-        if (r && r.abi.length > 0) return Promise.resolve(r);
-      } catch (error: any) {
-        if (error.status === 404) continue;
-        throw error;
-      }
+    loaders: ABILoader[];
+
+    constructor(loaders: ABILoader[]) {
+        this.loaders = loaders;
     }
-    return emptyContractResult;
-  }
 
-  async loadABI(address: string): Promise<any[]> {
-    for (const loader of this.loaders) {
-      const r = await loader.loadABI(address);
+    async getContract(address: string): Promise<ContractResult> {
+        for (const loader of this.loaders) {
+            try {
+                const r = await loader.getContract(address);
+                if (r && r.abi.length > 0) return Promise.resolve(r);
+            } catch (err: any) {
+                if (err.status === 404) continue;
+
+                throw new MultiABILoaderError("MultiABILoader getContract error: " + err.message, {
+                    context: { loader, address },
+                    cause: err,
+                });
+            }
+        }
+        return emptyContractResult;
+    }
 
-      // Return the first non-empty result
-      if (r.length > 0) return Promise.resolve(r);
+    async loadABI(address: string): Promise<any[]> {
+        for (const loader of this.loaders) {
+            try {
+                const r = await loader.loadABI(address);
+
+                // Return the first non-empty result
+                if (r.length > 0) return Promise.resolve(r);
+            } catch (err: any) {
+                throw new MultiABILoaderError("MultiABILoader loadABI error: " + err.message, {
+                    context: { loader, address },
+                    cause: err,
+                });
+            }
+        }
+        return Promise.resolve([]);
     }
-    return Promise.resolve([]);
-  }
 }
 
-export class EtherscanABILoader implements ABILoader {
-  apiKey?: string;
-  baseURL: string;
+export class MultiABILoaderError extends errors.LoaderError { };
```

Comment:
@roninjin10  We have `name` overrides for the base errors in `errors.ts`, but I didn't add them to all the leaf errors, think it's worth adding it everywhere?